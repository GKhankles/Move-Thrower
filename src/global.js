var advancedToggle = false;
var curGeneration = 3;
var weatherList = [];
var pkmn1 = null;
var pkmn2 = null;
var uid = "";