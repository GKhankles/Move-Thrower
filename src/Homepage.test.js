import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HomePage from './HomePage.jsx';
import global from './global.js';

var copyState = {
    "atkPkmnInfo": {
        "uid": "",
        "curPkmn": "Bulbasaur",
        "baseStats": {
            "HP": 45,
            "Atk": 49,
            "SpAtk": 65,
            "Def": 49,
            "SpDef": 65,
            "Spd": 45
        },
        "ivInfo": {
            "HP": 31,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "evInfo": {
            "HP": 0,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "totalStats": {
            "HP": 11,
            "Atk": 5,
            "Def": 5,
            "SpAtk": 6,
            "SpDef": 6,
            "Spd": 5
        },
        "level": 1,
        "nature": "Hardy",
        "status": "Healthy",
        "moves": [],
        "types": [
            "Grass",
            "Poison"
        ],
        "pkmnImg": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
        "pokemonList": [
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"]
        ],
        "isAdvanced": true,
        "readySwap": false,
        "savedName": "Insert Name",
        "display": false,
        "nameList": [],
        "pkmnList": []
    },
    "defPkmnInfo": {
        "uid": "",
        "curPkmn": "Bulbasaur",
        "baseStats": {
            "HP": 45,
            "Atk": 49,
            "SpAtk": 65,
            "Def": 49,
            "SpDef": 65,
            "Spd": 45
        },
        "ivInfo": {
            "HP": 0,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "evInfo": {
            "HP": 0,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "totalStats": {
            "HP": 11,
            "Atk": 5,
            "Def": 5,
            "SpAtk": 6,
            "SpDef": 6,
            "Spd": 5
        },
        "level": 1,
        "nature": "Hardy",
        "status": "Healthy",
        "moves": [],
        "types": [
            "Grass",
            "Poison"
        ],
        "pkmnImg": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
        "pokemonList": [
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"]
        ],
        "isAdvanced": true,
        "readySwap": false,
        "savedName": "Insert Name",
        "display": false,
        "nameList": [],
        "pkmnList": []
    },
    "isAdvanced": false,
    "calculatedMoves": null,
    "calculating": false,
    "weather": "Clear",
    "weatherList": null
};

var normalState = {
    "atkPkmnInfo": {
        "uid": "",
        "curPkmn": "Bulbasaur",
        "baseStats": {
            "HP": 45,
            "Atk": 49,
            "SpAtk": 65,
            "Def": 49,
            "SpDef": 65,
            "Spd": 45
        },
        "ivInfo": {
            "HP": 31,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "evInfo": {
            "HP": 0,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "totalStats": {
            "HP": 11,
            "Atk": 5,
            "Def": 5,
            "SpAtk": 6,
            "SpDef": 6,
            "Spd": 5
        },
        "level": 1,
        "nature": "Hardy",
        "status": "Healthy",
        "moves": [],
        "types": [
            "Grass",
            "Poison"
        ],
        "pkmnImg": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
        "pokemonList": [
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"]
        ],
        "isAdvanced": true,
        "readySwap": false,
        "savedName": "Insert Name",
        "display": false,
        "nameList": [],
        "pkmnList": []
    },
    "defPkmnInfo": {
        "uid": "",
        "curPkmn": "Bulbasaur",
        "baseStats": {
            "HP": 45,
            "Atk": 49,
            "SpAtk": 65,
            "Def": 49,
            "SpDef": 65,
            "Spd": 45
        },
        "ivInfo": {
            "HP": 0,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "evInfo": {
            "HP": 0,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "totalStats": {
            "HP": 11,
            "Atk": 5,
            "Def": 5,
            "SpAtk": 6,
            "SpDef": 6,
            "Spd": 5
        },
        "level": 1,
        "nature": "Hardy",
        "status": "Healthy",
        "moves": [],
        "types": [
            "Grass",
            "Poison"
        ],
        "pkmnImg": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
        "pokemonList": [
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"]
        ],
        "isAdvanced": true,
        "readySwap": false,
        "savedName": "Insert Name",
        "display": false,
        "nameList": [],
        "pkmnList": []
    },
    "isAdvanced": false,
    "calculatedMoves": null,
    "calculating": false,
    "weather": "Clear",
    "weatherList": null
}

test('HomePage Initial Snapshot Test', () => {
  const wrapper = shallow(<HomePage />);
  expect(wrapper).toMatchSnapshot();
});

/*test('HomePage ResetSettings Test', () => {
    const wrapper = shallow(<HomePage />);
    let newState = {...copyState};
    global.pkmn1 = {
        "uid": "",
        "curPkmn": "Bulbasaur",
        "baseStats": {
            "HP": 45,
            "Atk": 49,
            "SpAtk": 65,
            "Def": 49,
            "SpDef": 65,
            "Spd": 45
        },
        "ivInfo": {
            "HP": 0,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "evInfo": {
            "HP": 0,
            "Atk": 0,
            "SpAtk": 0,
            "Def": 0,
            "SpDef": 0,
            "Spd": 0
        },
        "totalStats": {
            "HP": 11,
            "Atk": 5,
            "Def": 5,
            "SpAtk": 6,
            "SpDef": 6,
            "Spd": 5
        },
        "level": 1,
        "nature": "Hardy",
        "status": "Healthy",
        "moves": [],
        "types": [
            "Grass",
            "Poison"
        ],
        "pkmnImg": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
        "pokemonList": [
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"],
            ["Bulbasaur"]
        ],
        "isAdvanced": true,
        "readySwap": false,
        "savedName": "Insert Name",
        "display": false,
        "nameList": [],
        "pkmnList": []
    };
    wrapper.setState({newState});
    wrapper.instance().changeGeneration({target: {value: 3}});
    wrapper.instance().resetSettings();
    expect(wrapper.state).toBe(normalState);
});*/

test('Test Create Move List', () => {
    const wrapper = shallow(<HomePage />);
    let newState = {...copyState};
    wrapper.setState({newState});
    wrapper.setState({ calculatedMoves: [{move_type: "Flying", move:"Fly", min_dmg:45, max_dmg:68}, {move_type: "Fairy", move:"Fly", min_dmg:45, max_dmg:75}, {move_type: "Rock", move:"Fly", min_dmg:45, max_dmg:89}, {move_type: "Bug", move:"Fly", min_dmg:45, max_dmg:122}] });
    wrapper.instance().createMoveList();
    expect(wrapper).toMatchSnapshot();
});

