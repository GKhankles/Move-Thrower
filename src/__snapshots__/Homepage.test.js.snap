// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Initial Snapshot Test 1`] = `
<div
  className="App"
  style={
    Object {
      "fontSize": 25,
    }
  }
>
  <div
    className="App-hcontainer"
  >
    <header
      className="App-header"
    >
      <h1>
        Pokemon Move Thrower!
      </h1>
    </header>
    <div
      className="App-login"
    >
      <SignIn
        preference={Object {}}
      />
    </div>
  </div>
  <div
    className="App-mid"
  >
    <div
      className="App-body"
    >
      <b>
        Attacking Pokemon
      </b>
      <Pokemon
        getPkmnInfo={[Function]}
        loc={0}
      />
    </div>
    <div
      className="App-body"
    >
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "fontSize": 18,
          }
        }
      >
        Switch Roles
      </button>
      <br />
      <br />
      <br />
      <button
        className="button"
        onClick={[Function]}
      >
        CALCULATE
      </button>
      <br />
      <br />
      <b>
        Advanced Options
      </b>
      <input
        onChange={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "height": 20,
            "width": 20,
          }
        }
        type="checkbox"
      />
      <br />
      <br />
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "fontSize": 18,
          }
        }
      >
        Reset Settings
      </button>
      <div
        className="advancedBody"
      />
      <br />
    </div>
    <div
      className="App-body"
    >
      <b>
        Defending Pokemon
      </b>
      <Pokemon
        getPkmnInfo={[Function]}
        loc={1}
      />
    </div>
  </div>
  <div
    className="App-body"
    style={
      Object {
        "display": "flex",
        "minWidth": "100%",
      }
    }
  >
    <div
      className="advancedBody2"
    />
    <b
      style={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      Recommended Moves
    </b>
  </div>
</div>
`;

exports[`Test Create Move List 1`] = `
<div
  className="App"
  style={
    Object {
      "fontSize": 25,
    }
  }
>
  <div
    className="App-hcontainer"
  >
    <header
      className="App-header"
    >
      <h1>
        Pokemon Move Thrower!
      </h1>
    </header>
    <div
      className="App-login"
    >
      <SignIn
        preference={Object {}}
      />
    </div>
  </div>
  <div
    className="App-mid"
  >
    <div
      className="App-body"
    >
      <b>
        Attacking Pokemon
      </b>
      <Pokemon
        getPkmnInfo={[Function]}
        loc={0}
      />
    </div>
    <div
      className="App-body"
    >
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "fontSize": 18,
          }
        }
      >
        Switch Roles
      </button>
      <br />
      <br />
      <br />
      <button
        className="button"
        onClick={[Function]}
      >
        CALCULATE
      </button>
      <br />
      <br />
      <b>
        Advanced Options
      </b>
      <input
        onChange={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "height": 20,
            "width": 20,
          }
        }
        type="checkbox"
      />
      <br />
      <br />
      <button
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "fontSize": 18,
          }
        }
      >
        Reset Settings
      </button>
      <div
        className="advancedBody"
      />
      <br />
    </div>
    <div
      className="App-body"
    >
      <b>
        Defending Pokemon
      </b>
      <Pokemon
        getPkmnInfo={[Function]}
        loc={1}
      />
    </div>
  </div>
  <div
    className="App-body"
    style={
      Object {
        "display": "flex",
        "minWidth": "100%",
      }
    }
  >
    <div
      className="advancedBody2"
    />
    <b
      style={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      Recommended Moves
    </b>
    <div
      className="moveList"
    >
      <div
        className="moveItem"
        style={
          Object {
            "backgroundImage": "linear-gradient(to right, rgba(169,143,243,1), rgba(169,143,243,1), white, rgba(169,143,243,1), rgba(169,143,243,1))",
          }
        }
      >
        <b>
          Fly
        </b>
        <b>
          45-68
        </b>
        <b>
          Flying
        </b>
      </div>
      <div
        className="moveItem"
        style={
          Object {
            "backgroundImage": "linear-gradient(to right, rgba(214,133,173,1), rgba(214,133,173,1), white, rgba(214,133,173,1), rgba(214,133,173,1))",
          }
        }
      >
        <b>
          Fly
        </b>
        <b>
          45-75
        </b>
        <b>
          Fairy
        </b>
      </div>
      <div
        className="moveItem"
        style={
          Object {
            "backgroundImage": "linear-gradient(to right, rgba(182,161,54,1), rgba(182,161,54,1), white, rgba(182,161,54,1), rgba(182,161,54,1))",
          }
        }
      >
        <b>
          Fly
        </b>
        <b>
          45-89
        </b>
        <b>
          Rock
        </b>
      </div>
      <div
        className="moveItem"
        style={
          Object {
            "backgroundImage": "linear-gradient(to right, rgba(166,185,26,1), rgba(166,185,26,1), white, rgba(166,185,26,1), rgba(166,185,26,1))",
          }
        }
      >
        <b>
          Fly
        </b>
        <b>
          45-122
        </b>
        <b>
          Bug
        </b>
      </div>
    </div>
  </div>
</div>
`;
